tests/publiccloud/sles4sap.pm

```diff
@@ -184,54 +184,81 @@ sub run {
-                    $instance->run_ssh_command(cmd => 'sudo sh -c \'until SAPHanaSR-showAttr | grep -q SOK; do sleep 1; done\'', timeout => $timeout);
+                    $instance->run_ssh_command(cmd => 'sudo sh -c \'until SAPHanaSR-showAttr | grep -q SOK; do sleep 1; done\'', timeout => $timeout, quiet => 1);
                     # Show HANA replication state
                     $self->run_cmd(cmd => 'SAPHanaSR-showAttr');
                 }
 
                 # Workaround bsc#1179529 on DRBD cluster (only for GCP)
-                if (is_gce and $cluster_type eq 'drbd' and $hostname =~ m/${cluster_type}01$/) {
+                if (is_gce and $cluster_type eq 'drbd') {
                     record_soft_failure 'bsc#1179529 - [ha-sap-terraform-deployments_v6] DRBD resource fails after reboot on GCE';
+
                     # Get the UUID of DRBD device
                     my $drbd_conf_file = '/etc/drbd.d/sapdata.res';
-                    my $drbd_device = $self->run_cmd(cmd => "awk '/[[:blank:]]+disk[[:blank:]]+/ { print \$NF }' $drbd_conf_file | sed -n '/\\/dev\\//s/;//p'", quiet => 1);
+                    my $drbd_device = $self->run_cmd(cmd => "awk '/[[:blank:]]+disk[[:blank:]]+\\/dev\\// { print \$NF }' $drbd_conf_file | sed 's/\\/dev\\/\\(.*\\);\$/\\1/'", quiet => 1);
                     die "DRBD device can't be found!" unless defined $drbd_device;
-                    my $drbd_uuid_device = $self->run_cmd(cmd => "blkid -o export $drbd_device | awk -F'=' '/^UUID=/ { print \$NF }'", quiet => 1);
-                    die "DRBD UUID can't be found!" unless defined $drbd_uuid_device;
-
-                    # Replace current device name by its UUID in config file and sync cluster conf
-                    $self->run_cmd(cmd => "sed -i '/[[:blank:]]*disk[[:blank:]]*\\/dev\\//s;\\(^[[:blank:]]*\\).*;\\1disk /dev/disk/by-uuid/${drbd_uuid_device}\\;;' $drbd_conf_file", quiet => 1);
+                    my $drbd_id_device = $self->run_cmd(cmd => "ls -l /dev/disk/by-id/ 2>/dev/null | awk '/\\/${drbd_device}\$/ { print \$9 }' | head -n1", quiet => 1);
+                    die "DRBD device ID can't be found!" unless defined $drbd_id_device;
+
+                    # All node should be configured
+                    my @nodes = split /\n/, $self->run_cmd(cmd => 'crm node show | sed -n \'/:[[:blank:]]*/s/(.*//p\'', quiet => 1);
+                    foreach my $node (@nodes) {
+                        # Replace current device name by its ID in config file and sync cluster conf
+                        $drbd_id_device =~ s/-0-part/-1-part/ unless $node eq $hostname;
+                        $self->run_cmd(cmd => "sed -i '/[[:blank:]]*disk[[:blank:]]*\\/dev\\//d' $drbd_conf_file", quiet => 1) if $node eq $hostname;
+                        $self->run_cmd(cmd => "sed -i '/on $node {/a\\disk /dev/disk/by-id/${drbd_id_device};' $drbd_conf_file", quiet => 1);
```

Comment:
Instead of one command to delete the lines with `/dev/` and another to add `$drbd_id_device` into the drbd configuration file, isn't it possible to simply replace `/dev/$drbd_device` for `$drbd_id_device` in a single sed command? Or am I missing something?

Reply:
No: the command remove the initial `disk /dev/...` entry which is unique and common to all node and add a new `disk /dev/...` line for each node, as the block device is not the same on both nodes.
So we have to delete the common line first, and add the specific line for each node in the correct section. So at the end we have 2 `disk /dev/...` lines.

Reply:
Thinking now that perhaps you did it like this due to the `/` in the strings. If that's the case, I would suggest:
```
$self->run_cmd(cmd => "sed -i 's|/dev/$drbd_device|$drbd_id_device|' $drbd_conf_file", quiet => 1);
```

Reply:
So, instead of a `disk` parameter for each node, the automation is adding a global `disk` parameter for the whole configuration, even when this device has different backing devices in each of the cluster nodes? Seems risky. :P